### DISCUSSION TITLE
Why does vault-configurer use root token?

### DISCUSSION THREAD

**Discussion by Ais8Ooz8** [bank-vaults/vault-operator#152] [Q&A] [answered]
If you try to revoke your root token and remove it from vault-unseal-keys, the following errors appear:

**"error configuring vault: unable to get key vault-root: key vault-root is not present in secret: vault-unseal-keys"**

**Comment by johnny990** [comment]
it uses root token to configure vault. I have similar security concerns about it, but didn't find proper solution

**Comment by Ais8Ooz8** [comment, marked as answer]
There is an option `storeRootToken` to disable root token storage

https://github.com/bank-vaults/vault-operator/blob/main/pkg/apis/vault/v1alpha1/vault_types.go

https://github.com/bank-vaults/vault-operator/blob/main/deploy/examples/cr-disabled-root-token-storage.yaml
